src/microbe_stage/systems/EngulfingSystem.cs

```diff
@@ -206,15 +206,17 @@ protected override void Update(float delta, in Entity entity)
 
             var compounds = entity.Get<CompoundStorage>().Compounds;
 
-            // Stop engulfing if out of ATP or if this is an engulfable that has been engulfed
+            // Stop engulfing if this is an engulfable that has been engulfed
             bool engulfed = false;
 
             if (entity.Has<Engulfable>())
             {
                 engulfed = entity.Get<Engulfable>().PhagocytosisStep != PhagocytosisPhase.None;
             }
 
-            if (compounds.TakeCompound(atp, cost) < cost - 0.001f || engulfed)
```

Comment:
This is actually a good catch that there's a duplicate engulf cost taken here as well as in osmoregulation and healing system. Probably should remove that other place that causes double the intended ATP drain rate.